### PR TITLE
ELSI: add v2.11 and dlaf variant

### PR DESCRIPTION
**Pull request by RMeli**
Add ELSI `2.11` and support for DLA-Future.

### CODE REVIEW

**Comment by RMeli** [comment]
@spackbot fix style

**Review by msimberg — DISMISSED** [review]
Hash confirmed. Looks good to me otherwise as well.

File: var/spack/repos/builtin/packages/elsi/package.py [hunk 1]
```diff
@@ -65,11 +68,18 @@ class Elsi(CMakePackage, CudaPackage):
...
         "internal_elpa_version",
-        default="2024",
-        values=("2024", "2023_11", "2023", "2021", "2020"),
+        default="2020",
+        values=(
+            "2020",
+            "2021",
+            conditional("2023", when="@:2.11"),
+            conditional("2023_11", when="@:2.11"),
+            conditional("2024", when="@:2.11"),
```
Comment (albestro) [inline var/spack/repos/builtin/packages/elsi/package.py]: Just FYI, this is equivalent

```suggestion
            conditional("2023", "2023_11", "2024", when="@:2.11"),
```

File: var/spack/repos/builtin/packages/elsi/package.py [hunk 2]
```diff
@@ -97,6 +110,12 @@ class Elsi(CMakePackage, CudaPackage):
     with when("+use_external_omm"):
         depends_on("omm")
         depends_on("matrix-switch")  # Direct dependency
+    with when("+dlaf"):
+        depends_on("dla-future-fortran")
+        conflicts("dla-future~cuda", when="+cuda")
+        conflicts("dla-future+cuda", when="~cuda")
+
+    conflicts("^pexsi@2:", when="@:2.11")
```
Comment (albestro) [inline var/spack/repos/builtin/packages/elsi/package.py]: is there any reason why you preferred to have this conflict not in related `when` scope? I was looking on how you dealt with `ntpoly` and I thought it looked the same use-case.

File: var/spack/repos/builtin/packages/elsi/package.py (hunk 1 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/elsi/package.py]: Thanks for pointing this out!

File: var/spack/repos/builtin/packages/elsi/package.py (hunk 2 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/elsi/package.py]: Not really. Before, `ntpoly` didn't have a `when` context, so I put both conflicts together at the bottom. Later I moved the `ntpoly` one in the newly created context, but did not move the `pexsi` one.

**Review by albestro — APPROVED** [review]

**Pull request merged**